### PR TITLE
Add support to `total` like `functional`

### PR DESCRIPTION
**Pull request by Robertorosmaninho**
Fixes: #667 

Create support for the `[total]` attribute to replace `[functional]` in the future.

### CODE REVIEW

**Review by dwightguth — APPROVED** [review]

**Comment by radumereuta** [comment]
How did you test if it works?
You can locally change the front-end: https://github.com/runtimeverification/k/blob/master/kernel/src/main/java/org/kframework/backend/kore/ModuleToKORE.java#L1486
and it will print `total` instead of `functional`. If all tests from `regression-new` passes, then it should be fine.

File: lib/ast/AST.cpp [hunk 1]
```diff
@@ -1233,6 +1234,7 @@ bool KOREAxiomDeclaration::isRequired() {
   return !attributes.count(ASSOC) && !attributes.count(COMM)
          && !attributes.count(IDEM) && !attributes.count(UNIT)
          && !attributes.count(FUNCTIONAL) && !attributes.count(CONSTRUCTOR)
+         && !attributes.count(TOTAL) && !attributes.count(CONSTRUCTOR)
```
Comment (Baltoli) [inline lib/ast/AST.cpp:1237]: This duplicates the Constructor check.

**Review by Baltoli — APPROVED** [review]
One minor comment; otherwise looks good to me if the frontend tests pass.

**Comment by Robertorosmaninho** [comment]
> How did you test if it works?
> You can locally change the front-end: https://github.com/runtimeverification/k/blob/master/kernel/src/main/java/org/kframework/backend/kore/ModuleToKORE.java#L1486
> and it will print `total` instead of `functional`. If all tests from `regression-new` passes, then it should be fine.

Good idea! Thanks, Radu! Changing this line, I got an error on test `issue-1760` without my modification, and the error was solved within it!

**Pull request merged**